poseidon/poseidon.go

```diff
@@ -63,13 +63,24 @@ func mix(state []*ff.Element, t int, m [][]*ff.Element) []*ff.Element {
 
 // HashWithState computes the Poseidon hash for the given inputs and initState
 func HashWithState(inpBI []*big.Int, initState *big.Int) (*big.Int, error) {
+	res, err := hashWithState(inpBI, initState, 1)
+	if err != nil {
+		return nil, err
+	}
+	return res[0], nil
+}
+
+func hashWithState(inpBI []*big.Int, initState *big.Int, nOuts int) ([]*big.Int, error) {
```

Comment:
Please rename `hashWithState` to `HashWithStateEx` (and in this way make it public).